build.cake

```diff
@@ -0,0 +1,183 @@
+            var sample = System.IO.Path.GetFileNameWithoutExtension(proj);
+            DisplayHeading("Building " + sample + " sample");
+            BuildProject(proj, configuration);
+        }
+        
+    });
+
+//////////////////////////////////////////////////////////////////////
+// TEST
+//////////////////////////////////////////////////////////////////////
+
+Task("Test")
+.IsDependentOn("Rebuild")
+.OnError(exception => ErrorDetail.Add(exception.Message))
+.Does(() =>
+    {
+        foreach(var proj in ProjList)
+        {
+            var bin = DirFrom(proj) + "/bin/";
+            var projName = System.IO.Path.GetFileNameWithoutExtension(proj);
+            var dllName = bin + projName + ".dll";
+
+            DisplayHeading("Testing " + projName + " sample");
+
+            try
+            {
+                NUnit3(dllName);
+            }
+            catch(CakeException ex)
+            {
+                ErrorDetail.Add("     * " + projName + " test failed.");
+            }
+
+        }
+    });
+
+//////////////////////////////////////////////////////////////////////
+// TEARDOWN TASK
+//////////////////////////////////////////////////////////////////////
+
+Teardown(() =>
+    {
+        CheckForError(ref ErrorDetail);
+    });
+
+void CheckForError(ref List<string> errorDetail)
+{
+    if(errorDetail.Count != 0)
+    {
+        var copyError = new List<string>();
+        copyError = errorDetail.Select(s => s).ToList();
+        errorDetail.Clear();
+        throw new Exception("One or more unit tests failed, breaking the build.\n"
+            + copyError.Aggregate((x,y) => x + "\n" + y));
+    }
+}
+
+//////////////////////////////////////////////////////////////////////
+// HELPER METHODS
+//////////////////////////////////////////////////////////////////////
+
+void BuildProject(string projPath, string configuration)
+{
+    if (IsRunningOnWindows())
+    {
+        MSBuild(projPath, new MSBuildSettings()
+            .SetConfiguration(configuration)
+            .SetMSBuildPlatform(MSBuildPlatform.Automatic)
+            .SetVerbosity(Verbosity.Minimal)
+            .SetNodeReuse(false));
+    }
+    else
+    {
+        XBuild(projPath, new XBuildSettings()
+            .WithTarget("Build")
+            .WithProperty("Configuration", configuration)
+            .SetVerbosity(Verbosity.Minimal));
+    }
+}
+
+string DirFrom(string filePath)
+{
+    return System.IO.Path.GetDirectoryName(filePath);
+}
+
+void DisplayHeading(string heading)
+{
+    Information("");
+    Information("----------------------------------------");
+    Information(heading);
+    Information("----------------------------------------");
+    Information("");
+}
+
+
+//////////////////////////////////////////////////////////////////////
+// TASK TARGETS
+//////////////////////////////////////////////////////////////////////
+
+Task("Rebuild")
+.IsDependentOn("Clean")
+.IsDependentOn("Build");
+
+Task("Default")
+.IsDependentOn("Rebuild");
```

Comment:
I think of command line as a UI. So if I type build it seems like that's
what it should do.
On Mar 25, 2016 5:40 AM, "Chris Maddock" <notifications@github.com> wrote:

> In build.cake
> <https://github.com/nunit/nunit-vb-samples/pull/4#discussion_r57440743>:
>
> > +    Information(heading);
> > +    Information("----------------------------------------");
> > +    Information("");
> > +}
> > +
> > +
> > +//////////////////////////////////////////////////////////////////////
> > +// TASK TARGETS
> > +//////////////////////////////////////////////////////////////////////
> > +
> > +Task("Rebuild")
> > +.IsDependentOn("Clean")
> > +.IsDependentOn("Build");
> > +
> > +Task("Default")
> > +.IsDependentOn("Rebuild");
>
> Had left "Appveyor" out as it seemed redundant for samples, but will put
> it in for consistency =)
>
> Our other scripts use Build as default
>
> I saw, although nunit/build.cake has the comment:
>
> Task("Default")
>     .IsDependentOn("Build"); // Rebuild?
>
> which was why I used Rebuild here. Seems preferable to me that if you run
> a build script which fails, you can't accidently then work on an old build.
> I should probably have highlighted that for discussion, slipped my mind,
> sorry.
>
> Also, "build" is the name of the command.
>
> Not sure what you mean there, sorry?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/nunit/nunit-vb-samples/pull/4/files/d35e601929a1460002ca662c63d49a1687ea38d0#r57440743>
>